### ISSUE TITLE
Classifiers should support gaps in class labels

### ISSUE THREAD

**Issue by twotabbies** [haifengl/smile#462]
## Expected behaviour

Often users are benerating training data based on new observations covering existing already-classified areas. For example, there exist global datasets of land cover classification (e.g., MODIS-derived classification https://developers.google.com/earth-engine/datasets/catalog/MODIS_051_MCD12Q1) and users may be interested in training on new observations, but in areas where some land cover types are not present (e.g., training on farmland in Iowa; there are no glaciers in Iowa). Smile currently doesn't allow this: there must be at least one instance of each class in the training data. There is also no allowance for structural gaps in the class label (again using MCD12Q1.051 as an example, "unclassified" is label 254). While it would be possible for users to map class labels before passing the data to Smile, this is error-prone.

I have prepared a PR that implements this mapping internally to Smile for most classifiers. The ones I haven't implemented it for are LDA, QDA and RDA as their public interface involves a "priori" array, and handling that in the presence of index remapping gets tricky.

## Actual behaviour

Smile's classifiers reject training sets with missing class labels.

**Comment by haifengl** [comment]
v2 has this feature now
